applications/machine_learning/src/events/ml_result_event.c

```diff
@@ -14,7 +14,7 @@ static void log_ml_result_event(const struct app_event_header *aeh)
 	const struct ml_result_event *event = cast_ml_result_event(aeh);
 
 	APP_EVENT_MANAGER_LOG(aeh, "%s val: %0.2f anomaly: %0.2f",
-			event->label, event->value, event->anomaly);
+			event->label, (double)event->value, (double)event->anomaly);
```

Comment:
why is this needed - aren't all float arguments promoted to double in such function calls anyway?

Reply:
Yes, they are. This results in a warning however (since https://github.com/zephyrproject-rtos/zephyr/pull/57154) as Zephyr enabled `-Wdouble-promotion` compiler flag, which gives a warning in such cases (implict promotion to double). Therefore, we either need to make the promotion explicit to avoid the warning, or find some other way to avoid implicit promotion.

Reply:
thanks. zephyr should have taken on java instead...